### PR TITLE
chore: add format GitHub action

### PR DESCRIPTION
**Pull request by mcansh**
example repo: https://github.com/mcansh/test-repo

### CODE REVIEW

File: .github/workflows/format.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Format
+        run: yarn format --if-present
```
Comment (mcansh) [inline .github/workflows/format.yml:34]: looks like yarn doesn't have a `--if-present` flag, so going to switch this to run with npm

**Review by FStell01 — COMMENTED** [review]
?

File: .github/workflows/format.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    strategy:
+      matrix:
+        node-version: [14.x]
+
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
```
Comment (MichaelDeBoey) [inline .github/workflows/format.yml:20]: `styfle/cancel-workflow-action`'s `access_token` is already defaulting to `github.token`
https://github.com/styfle/cancel-workflow-action/blob/dee858fff096b2cbcd4f845ce21871c50c5af452/action.yml#L15-L17

```suggestion
```

File: .github/workflows/format.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
```
Comment (MichaelDeBoey) [inline .github/workflows/format.yml:28]: No need to put this in a matrix, as it's only running on one version
```suggestion
      - name: Use Node.js
        uses: actions/setup-node@v2
        with:
          node-version: 14
```

File: .github/workflows/format.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    branches:
+      - main
+
+jobs:
+  prettier:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.x]
```
Comment (MichaelDeBoey) [inline .github/workflows/format.yml:14]:
```suggestion
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

File: .github/workflows/format.yml (hunk 2 above)
Comment (mcansh) [reply .github/workflows/format.yml:20]: oh cool! i just used the example in the readme 🤷

**Review by mjackson — APPROVED** [review]

File: .github/workflows/format.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Format
+        run: npm run format --if-present
+
+      - name: Commit
+        run: |
+          git config --local user.email "bot@remix.run"
```
Comment (mjackson) [inline .github/workflows/format.yml:32]:
```suggestion
          git config --local user.email "github-actions@remix.run"
```

File: .github/workflows/format.yml (hunk 3 above)
Comment (mcansh) [reply .github/workflows/format.yml]: i actually had with out a matrix, but i have no idea why i switched it use one haha

Comment (MichaelDeBoey) [reply .github/workflows/format.yml]: Haha those things happen 😛

**Pull request merged**